# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasUtils.cpp
# Commit: 904012057423
# Full Hash: 904012057423b23bf9a382da27a40869c6c00301
# Author: Ehsan Akhgari <ehsan@mozilla.com>
# Date: 2019-03-06 09:56:17
# Regressor Bug: 1532414
# File Overlap Count: 1
# Description:
#   Bug 1532414 - Pass a document/principal pointer to nsContentUtils::ShouldResistFingerprinting() callers in the canvas API; r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22002
# ==============================================================================

diff -r 152b9d59cb29 -r 904012057423 dom/canvas/CanvasUtils.cpp
--- a/dom/canvas/CanvasUtils.cpp	Tue Mar 05 23:13:23 2019 +0000
+++ b/dom/canvas/CanvasUtils.cpp	Tue Mar 05 23:20:08 2019 +0000
@@ -49,7 +49,7 @@
 bool IsImageExtractionAllowed(Document* aDocument, JSContext* aCx,
                               nsIPrincipal& aPrincipal) {
   // Do the rest of the checks only if privacy.resistFingerprinting is on.
-  if (!nsContentUtils::ShouldResistFingerprinting()) {
+  if (!nsContentUtils::ShouldResistFingerprinting(aDocument)) {
     return true;
   }
 